client/packages/system/src/Report/ListView/ListView.tsx

```diff
@@ -10,34 +10,56 @@ import {
   ReportContext,
   PrinterIcon,
   BaseButton,
+  useDialog,
+  BasicSpinner,
 } from '@openmsupply-client/common';
 import { JsonData } from '@openmsupply-client/programs';
 import { useReport, ReportRowFragment } from '../api';
 import { Toolbar } from './Toolbar';
 import { ReportArgumentsModal } from '../components/ReportArgumentsModal';
 
+type PrintingDialogProps = { isPrinting: boolean };
+const PrintingDialog: React.FC<PrintingDialogProps> = (
+  props: PrintingDialogProps
+) => {
+  const { isPrinting } = props;
+  const { Modal, showDialog, hideDialog } = useDialog();
+
+  useEffect(() => {
+    if (isPrinting) {
+      showDialog();
+    } else {
+      hideDialog();
+    }
+  }, [isPrinting]);
+
+  return (
+    <Modal title="" height={200}>
+      <BasicSpinner messageKey="messages.fetching-report-data"></BasicSpinner>
+    </Modal>
+  );
+};
+
 const PrintButton = ({
   report,
-  setArguments,
+  onReportSelected,
+  isPrinting,
 }: {
   report: ReportRowFragment;
-  setArguments: (report?: ReportRowFragment) => void;
+  onReportSelected: (report?: ReportRowFragment) => void;
+  isPrinting: boolean;
 }) => {
   const t = useTranslation();
-  const { print } = useReport.utils.print();
   const onClick = () => {
-    if (report.argumentSchema) {
```

Comment:
Is this if statement redundant ?

Reply:
Yes there is another onReportSelected callback that was doing the same... I noticed it because there were two different `useReport.utils.print()` usages that were used depending on if you have arguments or not and thus my dialog wasn't showing for one of the cases...